tangelo/toolboxes/molecular_computation/tests/test_rdms.py

```diff
@@ -14,39 +14,78 @@
 
 import os
 import unittest
+import json
+import numpy as np
+from numpy.testing import assert_allclose
 
 from openfermion.utils import load_operator
 
+from tangelo.toolboxes.measurements import RandomizedClassicalShadow
 from tangelo.toolboxes.operators import FermionOperator
-from tangelo.toolboxes.molecular_computation.rdms import energy_from_rdms
+from tangelo.toolboxes.molecular_computation.rdms import energy_from_rdms, compute_rdms
 
 # For openfermion.load_operator function.
 pwd_this_test = os.path.dirname(os.path.abspath(__file__))
 
+ferm_op_of = load_operator("H2_ferm_op.data", data_directory=pwd_this_test + "/data", plain_text=True)
+ferm_op = FermionOperator()
+ferm_op.__dict__ = ferm_op_of.__dict__
+ferm_op.n_spinorbitals = 4
+ferm_op.n_electrons = 2
+ferm_op.spin = 0
```

Comment:
What happens if you do
```python3
ferm_op = FermionOperator(n_spinorbitals=4, n_electrons=2, spin=0) + load_operator("H2_ferm_op.data", data_directory=pwd_this_test + "/data", plain_text=True)
```

Reply:
I get 
`TypeError: Cannot add invalid type to <class 'tangelo.toolboxes.operators.operators.FermionOperator'>.`
Should that be fixed somehow?

Reply:
Yes. If you look in the same file at QubitHamiltonian. You will need to add something like
```python3
    def __iadd__(self, other_hamiltonian):

        # Raise error if attributes are not the same across Hamiltonians. This
        # check is ignored if comparing to a QubitOperator or a bare
        # QubitHamiltonian.
        if self.mapping is not None and self.up_then_down is not None and \
                                other_hamiltonian.mapping is not None and \
                                other_hamiltonian.up_then_down is not None:

            if self.mapping.upper() != other_hamiltonian.mapping.upper():
                raise RuntimeError("Mapping must be the same for all QubitHamiltonians.")
            elif self.up_then_down != other_hamiltonian.up_then_down:
                raise RuntimeError("Spin ordering must be the same for all QubitHamiltonians.")

        return super(QubitOperator, self).__iadd__(other_hamiltonian)

    def __eq__(self, other_hamiltonian):

        # Additional checks for == operator. This check is ignored if comparing
        # to a QubitOperator or a bare QubitHamiltonian.
        if self.mapping is not None and self.up_then_down is not None and \
                                other_hamiltonian.mapping is not None and \
                                other_hamiltonian.up_then_down is not None:
            if (self.mapping.upper() != other_hamiltonian.mapping.upper()) or (self.up_then_down != other_hamiltonian.up_then_down):
                return False

        return super(QubitOperator, self).__eq__(other_hamiltonian)
```

Reply:
Added these and other methods. Now FermionOperator can be mixed with openfermion version intuitively. Added tests for those and fixed one for SecondQuantizedMolecule that was failing